### PR TITLE
Chore: Remove the deprecated Vector type

### PR DESCRIPTION
**Pull request by ryantxu**
Since Grafana 10, the vector interface is not necessary and people can simply use an Array.  For Grafana 11, we should remove the vector class so nothing new gets built that depends on the interface.

This PR does *not* remove the array prototype hack that lets normal arrays "implement" Vector -- this means that plugins trying to use the Vector calls `get(idx)` and `set(idx)` will continue to work.  However we should not keep this hack forever.

See https://github.com/grafana/grafana/pull/66187

# Release notice breaking change

The Vector interface that was deprecated in Grafana 10 has been removed in Grafana 11.  Rather than use an interface, with get/set function, you can simply use an Array.  Plugins compiled against older versions and depend on calling get/set will continue to work because the Array prototype still has a modified prototype.  This will be removed in the future

### CODE REVIEW

File: public/app/plugins/datasource/loki/sortDataFrame.ts [hunk 1]
```diff
@@ -85,10 +85,12 @@ export function sortDataFrameByTime(frame: DataFrame, dir: SortDirection): DataF
     ...rest,
     fields: fields.map((field) => ({
       ...field,
-      values: new SortedVector(field.values, index).toArray(),
-      nanos: field.nanos === undefined ? undefined : new SortedVector(field.nanos, index).toArray(),
+      values: sorted(field.values, index),
```
Comment (ryantxu) [inline public/app/plugins/datasource/loki/sortDataFrame.ts:88]: @gabor -- can you verify that this still does what you expect?  It is the only use of SortedVector

**Review by mckn — APPROVED** [review]
LGTM! Would be great if you could add something to the migration guide as well: https://grafana.com/developers/plugin-tools/migration-guides/update-from-grafana-versions/

File: packages/grafana-data/src/types/dataFrame.ts [hunk 2]
```diff
@@ -149,10 +148,8 @@ export interface Field<T = any, V = Vector<T>> {
 
   /**
    * The raw field values
-   * In Grafana 10, this accepts both simple arrays and the Vector interface
-   * In Grafana 11, the Vector interface will be removed
    */
-  values: V | T[];
```
Comment (ryantxu) [inline packages/grafana-data/src/types/dataFrame.ts:155]: ^^^ this is the most important change!

File: public/app/plugins/datasource/loki/sortDataFrame.ts [hunk 3]
```diff
@@ -85,10 +85,12 @@ export function sortDataFrameByTime(frame: DataFrame, dir: SortDirection): DataF
...
+      nanos: field.nanos === undefined ? undefined : sorted(field.nanos, index),
     })),
   };
+}
 
-  return frame;
+function sorted<T>(vals: T[], index: number[]): T[] {
+  return vals.map((v, idx) => vals[index[idx]]);
```
Comment (svennergr) [inline public/app/plugins/datasource/loki/sortDataFrame.ts:95]:
```suggestion
  return vals.map((_, idx) => vals[index[idx]]);
```

**Review by svennergr — APPROVED** [review]
Loki part LGTM

File: public/app/plugins/datasource/loki/sortDataFrame.ts (hunk 1 above)
Comment (svennergr) [reply public/app/plugins/datasource/loki/sortDataFrame.ts:88]: I did. LGTM!

**Pull request merged**